tagstudio/src/qt/ts_qt.py

```diff
@@ -1170,14 +1174,27 @@ def update_libs_list(self, path: Path | str):
         self.settings.endGroup()
         self.settings.sync()
 
-    def open_library(self, path: Path) -> LibraryStatus:
+    def open_library(self, path: Path) -> None:
         """Open a TagStudio library."""
         open_message: str = f'Opening Library "{str(path)}"...'
         self.main_window.landing_widget.set_status_label(open_message)
         self.main_window.statusbar.showMessage(open_message, 3)
         self.main_window.repaint()
 
-        open_status = self.lib.open_library(path)
+        open_status: LibraryStatus = self.lib.open_library(path)
+
+        # Migration is required
+        if open_status.message and open_status.message.startswith("[JSON]"):
```

Comment:
I am not sure using this message to figure out whether migration is required is the best idea, maybe a `migration_required` field in `LibraryStatus` would be better? (This way changing what looks like a log message doesn't break functionality)